packages/apputils-extension/src/announcements.ts

```diff
@@ -0,0 +1,235 @@
+          return;
+        }
+        const { id, tags }: { id?: string; tags?: Array<string> } = (change
+          .notification.options.data ?? {}) as any;
+        if ((tags ?? []).some(tag => ['news', 'update'].includes(tag)) && id) {
+          const update: { [k: string]: INewsState } = {};
+          update[id] = { seen: true, dismissed: true };
+          config.update(update as any);
+        }
+      });
+
+      const mustFetchNews = settings?.get('fetchNews').composite as
+        | boolean
+        | null;
+      if (mustFetchNews === null) {
+        const notificationId = Notification.emit(
+          trans.__(
+            'Do you want to receive official news from the JupyterLab team?'
+          ) +
+            '\n\n<a href="https://jupyterlab.readthedocs.io/en/latest/privacy_policies" target="_blank" rel="noreferrer">' +
+            trans.__('Please read the privacy policy.') +
+            '</a>',
+          'default',
+          {
+            autoClose: false,
+            actions: [
+              {
+                label: trans.__('Accept'),
+                callback: () => {
+                  Notification.dismiss(notificationId);
+                  config
+                    .update({})
+                    .then(() => fetchNews())
+                    .catch(reason => {
+                      console.error(`Failed to get the news:\n${reason}`);
+                    });
+                  settings?.set('fetchNews', true).catch(reason => {
+                    console.error(
+                      `Failed to save setting 'fetchNews':\n${reason}`
+                    );
+                  });
+                }
+              },
+              {
+                label: trans.__('Refuse'),
+                callback: () => {
+                  Notification.dismiss(notificationId);
+                  settings?.set('fetchNews', false).catch(reason => {
+                    console.error(
+                      `Failed to save setting 'fetchNews':\n${reason}`
+                    );
+                  });
+                }
+              }
+            ]
+          }
+        );
+      } else {
+        await fetchNews();
+      }
+
+      async function fetchNews() {
+        if (settings?.get('fetchNews').composite ?? false) {
+          try {
+            const response = await requestAPI<{
+              news: Notification.INotification[];
+            }>(NEWS_API_URL);
+
+            for (const { message, type, options } of response.news) {
+              // @ts-expect-error data has no index
+              const id = options.data!['id'] as string;
+              // Filter those notifications
+              const state = (config.data[id] as INewsState) ?? {
+                seen: false,
+                dismissed: false
+              };
+              if (!state.dismissed) {
+                options.actions = [
+                  {
+                    label: trans.__("Don't show me again"),
```

Comment:
As a general idea, I think we've tried to avoid contractions in menu items, etc. How about "Do not ask me again"?